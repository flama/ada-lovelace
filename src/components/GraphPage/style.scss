@import '../../config';

#graph-page {
  display: flex;
  flex-direction: column;

  > h3, .graph-head > h3 {
    @include wrap-mobile;
    margin-bottom: $space-xs;
    margin-top: $space-xl;

    @include large-up {
      @include grid(8);
      @include offsetLeft(2);
      margin-bottom: $space-xs;
    }
  }

  > .content {
    @include wrap-mobile;

    > p {
      margin-bottom: $space-xl;
      font-size: $font-size-base;
      line-height: $line-height-base;

      @include small-only {
        line-height: $line-height-roomy;
      }
    }

    @include large-up {
      @include grid(7);
      @include offsetLeft(2);
      display: flex;
      align-items: center;
    }

  }

  > .wrapper {
    padding-top: $space-xxl;
    padding-bottom: $space-xxl;
    background-color: rgba(#000, 0.2);

    > .graph-content {
      @include wrap;
    }

    @include large-up {
      padding-top: $space-xxl;
      padding-bottom: $space-xxl;
    }
  }

  .graph-head {
    display: flex;
    justify-content: space-between;
    align-items: baseline;

    .title {
      @include grid(7);

      @include small-only {
        font-size: $font-size-deca;
        line-height: $line-height-roomy;
        margin-left: 22px;
        margin-right: 22px;
      }

      display: block;
      margin-bottom: $space-l;
      margin-top: 0;
    }

     > .legend {
      > span {
        font-size: $font-size-centi;
        color: $light-purple;
        display: flex;
        align-items: center;

        &:before {
          content: '';
          display: inline-block;
          margin-right: $space-xxs;
          height: 11px;
          width: 11px;
          border-radius: 50%;
          background-color: $selected;
          text-align: center;
          color: #fff;
        }
      }

      @include small-only { display: none; }
    }
  }

  .information {
    font-size: $font-size-deci;
    line-height: $line-height-deci;

    @include large-up {
      padding-left: 0.5em;
    }
  }
}
