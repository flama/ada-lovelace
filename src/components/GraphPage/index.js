import React, { Component } from 'react'
import PropTypes from 'prop-types'

import FilterableScatterPlot from '../FilterableScatterPlot'
import DataLoader from '../DataLoader'

import './style.scss'

class GraphPage extends Component {
  constructor(props) {
    super(props)

    this.state = {
      dataList: {},
      options: {
        categories: []
      }
    }
  }

  fetchData = ({ dataList, options }) => {
    this.setState({ dataList, options })
  }

  styleTitle = position => {
    return `title -${position}`
  }

  render() {
    return (
      <section id="graph-page">
        <h3>wonderful women</h3>
        <div className="content">
          <p>{ `There's a woman behind every single change in our history â€” whether you know her
          or not. We have picked a few (for now) to try and show it, by sorting them into their
          influence areas and through time.` }</p>
        </div>
        <div className="graph-head">
          <h4 className={ this.styleTitle(this.props.titlePosition) }>
            Who are the wonderful women who helped shape the world?
          </h4>
          <div className="legend">
            <span>Women</span>
          </div>
        </div>
        <DataLoader fetchData={this.fetchData} />
        <FilterableScatterPlot dataList={this.state.dataList} options={this.state.options} />
        <p className="information">
          { `Information: We collected the information presented from public sources
            and do not take responsibility for the content in the external links.` }
        </p>
      </section>
    )
  }
}

GraphPage.propTypes = {
  titlePosition: PropTypes.oneOf(['top', 'inline'])
}

export default GraphPage;
