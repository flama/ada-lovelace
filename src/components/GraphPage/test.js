import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import GraphPage from '../GraphPage'

jest.mock('../DataLoader', () => {
  return props => {
    props.fetchData({
      dataList: [],
      options: {
        categories: ["potato"]
      }
    })
    return <div />
  }
})

describe("the graph page", () => {
  it("should render correctly", () => {
    const tree = shallow(<GraphPage />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
