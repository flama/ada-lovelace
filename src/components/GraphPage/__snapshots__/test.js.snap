// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the graph page should render correctly 1`] = `
<section
  id="graph-page"
>
  <h3>
    wonderful women
  </h3>
  <div
    className="content"
  >
    <p>
      There's a woman behind every single change in our history â€” whether you know her
                or not. We have picked a few (for now) to try and show it, by sorting them into their
                influence areas and through time.
    </p>
  </div>
  <div
    className="graph-head"
  >
    <h4
      className="title -undefined"
    >
      Who are the wonderful women who helped shape the world?
    </h4>
    <div
      className="legend"
    >
      <span>
        Women
      </span>
    </div>
  </div>
  <Component
    fetchData={[Function]}
  />
  <FilterableScatterPlot
    dataList={Object {}}
    options={
      Object {
        "categories": Array [],
      }
    }
  />
  <p
    className="information"
  >
    Information: We collected the information presented from public sources
                and do not take responsibility for the content in the external links.
  </p>
</section>
`;
